test/server/cards/02_SHD/upgrades/RichReward.spec.ts

```diff
@@ -0,0 +1,95 @@
+describe('Rich Reward', function() {
+    integration(function(contextRef) {
+        describe('Rich Reward\'s Bounty ability', function() {
+            it('should give Experience token up to two units', function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: [
+                            { card: 'battlefield-marine', upgrades: ['rich-reward'] },
+                            { card: 'atst' },
+                            { card: 'yoda#old-master' }
+                        ]
+                    },
+                    player2: {
+                        groundArena: ['wampa'],
+                        spaceArena: ['concord-dawn-interceptors']
+                    }
+                });
+
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.battlefieldMarine);
+                context.player1.clickCard(context.wampa);
+
+                expect(context.player2).toBeAbleToSelectExactly([context.wampa, context.atst, context.yoda, context.concordDawnInterceptors]);
+                expect(context.player2).toHavePassAbilityButton();
+                expect(context.player2).toHavePrompt('Select 2 cards');
+                context.player2.clickCard(context.wampa);
+                context.player2.clickCard(context.concordDawnInterceptors);
+                context.player2.clickPrompt('Done');
+                expect(context.wampa).toHaveExactUpgradeNames(['experience']);
+                expect(context.concordDawnInterceptors).toHaveExactUpgradeNames(['experience']);
+                expect(context.player2).toBeActivePlayer();
+            });
+
+            it('should give Experience token to one unit', function () {
```

Comment:
We haven't used `TargetMode.UpTo` a lot, would you mind adding an extra test confirming that if there are no available targets it just passes (can use `expect(context.player2).toBeActivePlayer();` to confirm that player1 got no prompt)